lib/manager/cocoapods/artifacts.ts

```diff
@@ -0,0 +1,171 @@
+import upath from 'upath';
+import fs from 'fs-extra';
+import { hrtime } from 'process';
+import { ExecOptions } from 'child_process';
+import { platform } from '../../platform';
+import { exec, ExecResult } from '../../util/exec';
+import { logger } from '../../logger';
+import { UpdateArtifactsConfig, UpdateArtifactsResult } from '../common';
+
+async function tryExec(
+  cmd: string,
+  tag: string,
+  opts: ExecOptions
+): Promise<ExecResult> {
+  let result = null;
+  try {
+    result = await exec(cmd, opts);
+  } catch (err) {
+    if (tag !== 'latest' && err.message.startsWith('Unable to find image')) {
+      result = await tryExec(cmd.replace(`:${tag}`, ':latest'), 'latest', opts);
+    } else throw err;
+  }
+  return result;
+}
+
+export async function updateArtifacts(
+  packageFileName: string,
+  updatedDeps: string[],
+  newPackageFileContent: string,
+  config: UpdateArtifactsConfig
+): Promise<UpdateArtifactsResult[] | null> {
+  await logger.debug(`cocoapods.getArtifacts(${packageFileName})`);
+
+  if (updatedDeps.length < 1) {
+    logger.debug('CocoaPods: empty update - returning null');
+    return null;
+  }
+
+  if (!config.localDir) {
+    logger.debug('CocoaPods: no local dir specified');
+    return null;
+  }
+
+  const packageFileAbsolutePath = upath.join(config.localDir, packageFileName);
+  const cwd = upath.dirname(packageFileAbsolutePath);
+
+  const lockFileName = upath.join(
+    upath.dirname(packageFileName),
+    'Podfile.lock'
+  );
+  const lockFileAbsolutePath = upath.join(cwd, 'Podfile.lock');
+
+  try {
+    await fs.outputFile(packageFileAbsolutePath, newPackageFileContent);
+  } catch (err) {
+    logger.warn({ err }, 'Podfile could not be written');
+    return [
+      {
+        lockFileError: {
+          lockFile: lockFileName,
+          stderr: err.message,
+        },
+      },
+    ];
+  }
+
+  const existingLockFileContent = await platform.getFile(lockFileName);
+  if (!existingLockFileContent) {
+    logger.debug(`Lockfile not found: ${lockFileName}`);
+    return null;
+  }
+
+  const cmdParts = [];
+  let cocoapodsVersion: string = null;
+  if (config.binarySource === 'docker') {
+    cmdParts.push('docker run --rm');
+
+    if (config.dockerUser) cmdParts.push(`--user=${config.dockerUser}`);
+
+    cmdParts.push(`-v ${cwd}:${cwd}`);
+    cmdParts.push(`-w ${cwd}`);
+    cmdParts.push(`--entrypoint pod`);
+
+    const match = existingLockFileContent.match(
+      /^COCOAPODS: (?<cocoapodsVersion>.*)$/m
+    ) || { groups: { cocoapodsVersion: 'latest' } };
+    cocoapodsVersion = match.groups.cocoapodsVersion;
+    cmdParts.push(`renovate/cocoapods:${cocoapodsVersion}`);
```

Comment:
Somehow related to #2565

Reply:
We ideally should have a more dynamic approach like we do with Ruby, where we fall back to `latest` if we can't match directly

Reply:
@zharinov this: https://github.com/renovatebot/renovate/blob/0eb12ce56ff75683872474523cd2433eb61c04ee/lib/manager/bundler/artifacts.ts#L66-L86